nixos/modules/services/networking/frp-client.nix

```diff
@@ -0,0 +1,72 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.frp-client;
+  settingsFormat = pkgs.formats.ini { };
+  configFile = settingsFormat.generate "frpc.ini" cfg.settings;
+in
+{
+  options = {
+    services.frp-client = {
+      enable = mkEnableOption (lib.mdDoc "Frp client configuration");
+
+      package = mkPackageOptionMD pkgs "frp" { };
+
+      settings = mkOption {
+        type = settingsFormat.type;
+        default = { };
+        description = lib.mdDoc ''
+          Frp client configuration, for configuration options see the example on [github](https://github.com/fatedier/frp/blob/dev/conf/frpc_full.ini)
+        '';
+        example = literalExpression ''
+          {
+            common = {
+              server_addr = "x.x.x.x";
+              server_port = 7000;
+            };
+          }
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services = {
+      frp-client = {
+        wants = [ "network.target" ];
+        after = [ "network.target" ];
+        wantedBy = [ "multi-user.target" ];
+        description = "A fast reverse proxy client.";
+        serviceConfig = {
+          Type = "simple";
+          Restart = "on-failure";
+          RestartSec = 15;
+          ExecStart = "${cfg.package}/bin/frpc -c ${configFile}";
+          # Hardening
+          CapabilityBoundingSet = "";
+          NoNewPrivileges = true;
+          ProtectSystem = "strict";
+          PrivateTmp = true;
+          PrivateDevices = true;
+          PrivateUsers = true;
+          ProtectHostname = true;
+          ProtectClock = true;
+          ProtectKernelTunables = true;
+          ProtectKernelModules = true;
+          ProtectKernelLogs = true;
+          ProtectControlGroups = true;
+          RestrictAddressFamilies = [ "AF_UNIX" "AF_INET AF_INET6" ];
```

Comment:
`"AF_INET AF_INET6"` may generate a working service file, but it is wrong from the perspective of nix lang.  I believe its a typoe.

Reply:
I believe this is a typo by the original author and will be corrected later.